### PR TITLE
Added validity proofs for eq, ge and aggregate proof

### PR DESCRIPTION
**Pull request by aritroCoder**

### CODE REVIEW

File: spec/data_flow_presentation_verify.md [hunk 1]
```diff
@@ -104,7 +105,7 @@ $$\widehat{T_6} \leftarrow  D^{\widehat{r''}}\cdot g^{-\widehat{m'}} \widetilde{
 $$\widehat{T_7} \leftarrow \left(\frac{e(pk\cdot\mathcal{G},\mathcal{S})}{e(g,g')}\right)^{c_H}\cdot e(pk\cdot \mathcal{G},\widehat{h})^{\widehat{r''}}\cdot e(\widetilde{h},\widehat{h})^{-\widehat{m'}}\cdot e(\widetilde{h},\mathcal{S})^{\widehat{r}}$$
 $$\widehat{T_8} \leftarrow \left(\frac{e(\mathcal{G},u)}{e(g,\mathcal{U})}\right)^{c_H}\cdot e(\widetilde{h},u)^{\widehat{r}}\cdot e(1/g,\widehat{h})^{\widehat{r'''}}$$
 
-Then all these values are added to $\widehat{T}$. This is then added with the validity proof which when hashed with $\mathcal{C}$ and $n_1$(recieved from [[ref: holder]]) re constructs the challenge hash $\widehat{c_H}$
+Then all these values are added to $\widehat{T}$. This is then added with the validity proof which when hashed with $\mathcal{C}$ and $n_1$(recieved from [[ref: holder]]) re constructs the challenge hash $\widehat{c_H}$.
```
Comment (swcurran) [inline spec/data_flow_presentation_verify.md:108]: Typo -- `received`

**Comment by swcurran** [comment]
I didn't understand how the validity proof that is processed as per this section is combined with the non-revocation proofs that are also in the presentation.  There will be one of those per source credential that have to be processed.

File: spec/data_flow_presentation_verify.md [hunk 2]
```diff
@@ -29,43 +29,44 @@ verification process, as noted below. Finally, an important part of the
...
-the predicate is described here.
-
-::: todo
-
-Add the aggregate_proof verification process
-
-:::
+#### Verify Validity Proofs
+
+An AnoncCreds validity proof is the combination of both equality and inequality predicate proofs. The validity proof is bound to the primary credential by the $\widehat{m_2}$ value that is presented in both proofs. The validity proof is verified by the following steps:
```
[the remark this answers is not in the document]
Comment (aritroCoder) [reply spec/data_flow_presentation_verify.md:34]: That would be good

**Comment by aritroCoder** [comment]
> I didn't understand how the validity proof that is processed as per this section is combined with the non-revocation proofs that are also in the presentation. There will be one of those per source credential that have to be processed.

To verify the proof, we need to regenerate the challenge hash at verifier side and match it with the one recieved from prover. The challenge hash is computed by concatinating `\mathcal{C}`, `\mathcal{T}` and nonce `n_1`. `\mathcal{C}` and `n_1` are sent by the holder to verifier, and verfier has to compute `\mathcal{T}` by using the NRP and validity proofs. That's why at the end of both NRP and validity proofs you would find the line "`add these values to \hat{T}`", which basically means we are regenerating `\mathcal{T}`.

**Comment by mikelodder7** [comment]
> > I didn't understand how the validity proof that is processed as per this section is combined with the non-revocation proofs that are also in the presentation. There will be one of those per source credential that have to be processed.
> 
> To verify the proof, we need to regenerate the challenge hash at verifier side and match it with the one recieved from prover. The challenge hash is computed by concatinating `\mathcal{C}`, `\mathcal{T}` and nonce `n_1`. `\mathcal{C}` and `n_1` are sent by the holder to verifier, and verfier has to compute `\mathcal{T}` by using the NRP and validity proofs. That's why at the end of both NRP and validity proofs you would find the line "`add these values to \hat{T}`", which basically means we are regenerating `\mathcal{T}`.

Correct

**Comment by swcurran** [comment]
Reopend pull request accidentally closed.  @mikelodder7 — could you please add your approval?  Thanks!

**Review by mikelodder7 — APPROVED** [review]

**Pull request merged**